adapters/goal.ts

```diff
@@ -1,49 +1,60 @@
-import axios from 'axios'
-import * as goalAdapter from './goal.transformer'
-import { type Goal } from '~/interfaces/Goal'
-import { type PostGoal } from '~/interfaces/PostGoal'
-import { API } from '~/constants/api'
+import axios from "axios";
+import * as goalAdapter from "./goal.transformer";
+import { type Goal } from "~/interfaces/Goal";
+import { type PostGoal } from "~/interfaces/PostGoal";
+import { API } from "../constants/api";
 
 const goalSiteConfig = {
   headers: {
-    'Content-Type': 'application/vnd.api+json'
-  }
-}
+    "Content-Type": "application/vnd.api+json",
+  },
+};
 
-export const fetchGoals = async (): Promise<Goal []> => {
+export const fetchGoals = async (): Promise<Goal[]> => {
   const goals: Goal[] = await axios
     .get(`${API.GOAL_BASE_URL}/goal/`)
-    .then(res => goalAdapter.transformGoalsFromApi(res.data.data))
-    .catch(error => {
-      throw new Error(error)
-    })
+    .then((res) => goalAdapter.transformGoalsFromApi(res.data.data))
+    .catch((error) => {
+      throw new Error(error);
+    });
 
-  return goals
-}
+  return goals;
+};
 
 export const addGoal = async (goal: PostGoal): Promise<Goal> => {
```

Comment:
here function is not checking the status code of the axios response. This means that if the axios request fails, the addGoal() function will return the goalResponse object, which may not be a Goal object.